pkg/registryclient/client.go

```diff
@@ -183,12 +182,23 @@ func WithTracing() Option {
 }
 
 // BuildRemoteOption builds remote.Option based on client.
-func (c *client) BuildRemoteOption(ctx context.Context) remote.Option {
-	return remote.WithRemoteOptions(
-		gcrremote.WithAuthFromKeychain(c.keychain),
-		gcrremote.WithTransport(c.transport),
-		gcrremote.WithContext(ctx),
-	)
+func (c *client) BuildRemoteOption(ctx context.Context) []gcrremote.Option {
+	remoteOpts := []gcrremote.Option{}
+	remoteOpts = append(remoteOpts, gcrremote.WithAuthFromKeychain(c.keychain))
+	remoteOpts = append(remoteOpts, gcrremote.WithTransport(c.transport))
+	remoteOpts = append(remoteOpts, gcrremote.WithContext(ctx))
+
+	pusher, err := gcrremote.NewPusher(remoteOpts...)
+	if err == nil {
+		remoteOpts = append(remoteOpts, gcrremote.Reuse(pusher))
+	}
+
+	puller, err := gcrremote.NewPuller(remoteOpts...)
+	if err == nil {
+		remoteOpts = append(remoteOpts, gcrremote.Reuse(puller))
```

Comment:
This won't work with cosign 2.0 as cosign 2.0.2 does not use the latest version of gcr, so they do not support remote.Reuse yet. You should re-add the cosign opts function and add this logic to something like a "gcrRemoteOpts".